### PR TITLE
Use load method to add tools

### PR DESCRIPTION
**Pull request by njhale**
- **enhance: use load method to add tools**
- **chore: bump node-gptscript to get load method**

Addresses #80

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,8 +0,0 @@
-node_modules
```
Comment (StrongMonkey) [inline .dockerignore:1]: Unintentional?

File: server/app.mjs [hunk 2]
```diff
@@ -248,54 +248,15 @@ const mount = async (
       }
     }
 
-    /*
```
Comment (StrongMonkey) [inline server/app.mjs:251]: 💯

File: server/app.mjs [hunk 3]
```diff
@@ -532,3 +454,85 @@ function gptscriptConfigPath() {
...
+      return {
+        function: {
+          toolID: key,
+          name: toolName,
+          description: tool.description,
+          parameters: args,
+        },
+      };
+    })
+    .filter((tool) => tool !== null);
```
Comment (StrongMonkey) [inline server/app.mjs:537]: Nit: you can just add an additional filter 
```
.filter(([key, tool]) => return tool.instructions))
```

**Review by StrongMonkey — APPROVED** [review]

File: .dockerignore (hunk 1 above)
Comment (njhale) [reply .dockerignore:1]: do we really need a `.dockerignore` if we're not packaging this as a container image?

File: server/app.mjs [hunk 4]
```diff
@@ -532,3 +454,85 @@ function gptscriptConfigPath() {
...
+  }
+  tool = tool.replace(/^sys\./, '');
+
+  if (validToolName.test(tool)) {
+    return tool;
+  }
+
+  if (tool.length > 55) {
+    tool = tool.slice(0, 55);
+  }
```
Comment (g-linville) [inline server/app.mjs:488]: Just curious, why are you capping it at 55 here instead of 64?

File: server/app.mjs (hunk 3 above)
Comment (njhale) [reply server/app.mjs]: good call! updated.

**Review by g-linville — APPROVED** [review]

File: .dockerignore (hunk 1 above)
Comment (g-linville) [reply .dockerignore:1]: Yes because that means we ignore Docker /s

File: server/app.mjs (hunk 4 above)
Comment (njhale) [reply server/app.mjs:488]: To match the logic in gptscript: https://github.com/gptscript-ai/gptscript/blob/1e08c1c2149112ac06d6605d471477955b45a3d0/pkg/types/toolname.go#L38

File: .dockerignore (hunk 1 above)
Comment (StrongMonkey) [reply .dockerignore:1]: yeah it is fine if we want to delete

**Pull request merged**